controller/env/appenv.go

```diff
@@ -243,34 +245,62 @@ func (ae *AppEnv) ValidateServiceAccessToken(token string, apiSessionId *string)
 	return serviceAccessClaims, nil
 }
 
-func (ae *AppEnv) GetServerCert() (serverCert *tls.Certificate, kid string, signingMethod jwt.SigningMethod) {
-	return ae.ServerCert, ae.serverSigner.KeyId(), ae.serverSigner.SigningMethod()
+// GetClientApiDefaultTlsJwtSigner returns the default JWT signer for client API operations.
+func (ae *AppEnv) GetClientApiDefaultTlsJwtSigner() *jwtsigner.TlsJwtSigner {
+	return ae.clientApiDefaultSigner
 }
 
+// GetRootTlsJwtSigner creates and returns a JWT signer using the root server certificate.
+func (ae *AppEnv) GetRootTlsJwtSigner() *jwtsigner.TlsJwtSigner {
```

Comment:
this function strikes me to be the sorta function that should be on the `*identity.TokenId` itself, no?

Reply:
It can't. The identity library parses one identity section and is unaware of other sections. It has no concept of the root controller identity. Just the identity it loads.